### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by garciajordy**
**Milestone-3:**
-Create a Board class
-Create a Game class with the logic behind the game
-Make the Board class inherit from the Game class
-Make sure it is IMPOSSIBLE to cheat.
-Require all the files in the main file

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative 'logic'
+# Board class to create a board
+class Board < Game
+  def display
+    print " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} \n",
```
Comment (umairarshadbutt) [inline lib/board.rb]: Remove the print from board.rb and move into main.rb

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @garciajordy and @JulianaOsemeke 
Such a remarkable project you made :+1: ! Awesome README :100: !
We, however, do have a few changes to carry out before we proceed (see review comment).

##

Kindly work on the suggested changes 👇.
- [x] All game-logic related resources (classes and/or modules) are kept in the lib directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
 
- [x] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

File: lib/board.rb (hunk 1 above)
Comment (garciajordy) [reply lib/board.rb]: Removed the print like requested

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @garciajordy and @JulianaOsemeke 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

 For this project, we have to make sure nothing is messed up. Kindly compare the screenshots below 👇 and work on the suggested changes.
- [x] As I explained that All game-logic-related resources (classes and/or modules) are kept in the lib directory. (screenshot from Pathwright). It means to remove the User class from main.rb.
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

File: lib/user.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(name)
+    name += "\'s" if name[-1] != 's'
+    @name = name
+  end
+
+  def mark
+    @marker
+  end
+end
```
Comment (pacyL2K19) [inline lib/user.rb]: Nice to have the class in a separated file :+1:

**Review by pacyL2K19 — CHANGES_REQUESTED** [review]
# Status: Changes Required

Hello @JulianaOsemeke and @garciajordy 
It seems like you worked hard to fit all the requirement, and the game logic is really fine, I liked the way you displayed the board :+1: 
However, I encountered an annoying infinite loop while testing the `Drawing move (See the screenshot) :arrow_down: 
warningTIcici
> 
Please consider it as an improvement of your code and submit again for another review
Feel free to reach out to me if any question 
>
Happy coding :desktop_computer:

File: lib/user.rb (hunk 2 above)
Comment (garciajordy) [reply lib/user.rb]: We actually resolved it on the moment you were reviewing it.. sorry about that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested image

Hello @JulianaOsemeke and @garciajordy I can see you are putting many efforts while
working on this project you need to be commended👏👏 

- User interface is awesome
- All combinations vertical, horizontal, and diagonal are showing  a winner when elements
are equal
- PR is professional
- Classes are in differents files

However, there is a small change you need to make to be able to move on

- Validate your user input so that it can't accept an empty name
- FIx Rubocop errors

Kindly fix that and submit another code review

# Keep Working Hard image

All the best,

Lydia Ingabire

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested image
Good job team for implementing some of the feedback 👏👏 but keep your head up
and work on these small changes

- you still have rubocop errors
shott
- try to add an error message when a user did not provide input to make it easier
for the user to know what is going on

## Fix that and submit for another code review and am sure you will be ready to move on

best,

Lydia Ingabire

**Comment by rukundoeric** [comment]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯
  - [X] Game logic is working exactly as it should. 💯
ls

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**